DOCS audit attributes for API Key authn (#60033)
elastic/elasticsearch: x-pack/docs/en/security/auditing/event-types.asciidoc

that have been previously described:

* `authentication_success`:

>   `realm`              ::   The name of the realm that successfully authenticated the user,
>                             or the special value of `_es_api_key` if authentication is
>                             performed using an API Key. This is a shorthand attribute
>                             for the same information that is described by the `user.realm`,
>                             `user.run_by.realm` and `authentication.type` attributes.

Reviewer: ```suggestion
  `realm`              ::   The name of the realm that successfully authenticated the user.
                            If authenticated using an API key, this is the special value of
                            `_es_api_key`. This is a shorthand attribute
                            for the same information that is described by the `user.realm`,
                            `user.run_by.realm` and `authentication.type` attributes.
```

  `user.name`          ::   The name of the _effective_ user. This is usually the
                            same as the _authenticated_ user, but if using the
                            <<run-as-privilege, run as authorization functionality>>
                            this instead denotes the name of the  _impersonated_ user.

>                             If authentication is performed using an API Key this attribute
>                             designates the name of the owner user of the key.
>   `user.realm`         ::   The realm name that the _effective_ user belongs to. If authentication
>                             is performed using an API Key this attribute designates the realm name
>                             that the owner of the key belongs to.

Reviewer: ```suggestion
                            If authenticated using an API key, this is
                            the name of the API key owner.
  `user.realm`         ::   Name of the the realm to which the _effective_ user 
                            belongs. If authenticated using an API key, this is
                            the name of the realm to which the API key owner belongs.
```

  `user.run_by.name`   ::   This attribute is present only if the request is
                            using the <<run-as-privilege, run as authorization functionality>>
                            and denotes the name of the  _authenticated_ user, 
                            which is also known as the _impersonator_.

>   `user.run_by.realm`  ::   This attribute is present only if the request is
>                             using the <<run-as-privilege, run as authorization functionality>>
>                             and denotes the name of the realm that the _authenticated_
>                             (_impersonator_) user belongs to.
>   `authentication.type`::   One of `REALM`, `API_KEY`, `TOKEN`, `ANONYMOUS` or `INTERNAL`.
>   `api_key.id`         ::   This attribute is present only if authentication is performed using an
>                             API Key, and contains the value for the key id that is returned in response
>                             to the <<security-api-create-api-key, Create API Key>> call.
>   `api_key.name`       ::   This attribute is present only if authentication is performed using an
>                             API Key, and contains the value for the key name given in the request
>                             to <<security-api-create-api-key, Create API Keys>>

Reviewer: ```suggestion
  `user.run_by.realm`  ::   Name of the realm to which the _authenticated_
                            (_impersonator_) user belongs.
                            This attribute is provided only if the request
                            uses the <<run-as-privilege, run as authorization functionality>>.
  `authentication.type`::   Method used to authenticate the user.
                            Possible values are `REALM`, `API_KEY`, `TOKEN`, `ANONYMOUS` or `INTERNAL`.
  `api_key.id`         ::   API key ID returned by the <<security-api-create-api-key,create API key>> request.
                            This attribute is only provided for authentication using an API key.
  `api_key.name`       ::   API key name provided in the <<security-api-create-api-key,create API key>> request.
                            This attribute is only provided for authentication using an API key.
```

                             action is granted or denied.
  `user.run_as.realm`  ::    The realm name of that the _impersonated_ user belongs to.

* `access_granted` and `access_denied`:

>   `user.roles`         ::    The role names of the user as an array. If authentication is
>                              performed using an API Key this attribute designates the
>                              role names of the owner user of the key.

Reviewer: ```suggestion
  `user.roles`         ::    The role names of the user as an array. If authenticated
                             using an API key, this contains the
                             role names of the API key owner.
```

  `user.name`          ::    The name of the _effective_ user. This is usually the
                             same as the _authenticated_ user, but if using the
                             <<run-as-privilege, run as authorization functionality>>
                             this instead denotes the name of the  _impersonated_ user.

>                              If authentication is performed using an API Key this attribute
>                              designates the name of the owner user of the key.

Reviewer: ```suggestion
                             If authenticated using an API key, this is
                             the name of the API key owner.
```

>   `user.realm`         ::    The realm name that the _effective_ user belongs to. If authentication
>                              is performed using an API Key this attribute designates the realm name
>                              that the owner of the key belongs to.

Reviewer: ```suggestion
  `user.realm`         ::   Name of the the realm to which the _effective_ user 
                            belongs. If authenticated using an API key, this is
                            the name of the realm to which the API key owner belongs.
```

                             using the <<run-as-privilege, run as authorization functionality>>
                             and denotes the name of the realm that the _authenticated_
                             (_impersonator_) user belongs to.

>   `authentication.type`::    One of `REALM`, `API_KEY`, `TOKEN`, `ANONYMOUS` or `INTERNAL`.
>   `api_key.id`         ::    This attribute is present only if authentication is performed using an
>                              API Key, and contains the value for the key id that is returned in response
>                              to the <<security-api-create-api-key, Create API Key>> call.
>   `api_key.name`       ::    This attribute is present only if authentication is performed using an
>                              API Key, and contains the value for the key name given in the request
>                              to <<security-api-create-api-key, Create API Keys>>

Reviewer: ```suggestion
  `authentication.type`::   Method used to authenticate the user.
                            Possible values are `REALM`, `API_KEY`, `TOKEN`, `ANONYMOUS` or `INTERNAL`.
  `api_key.id`         ::   API key ID returned by the <<security-api-create-api-key,create API key>> request.
                            This attribute is only provided for authentication using an API key.
  `api_key.name`       ::   API key name provided in the <<security-api-create-api-key,create API key>> request.
                            This attribute is only provided for authentication using an API key.
```
